src/commons/api/aspa.rs

```diff
@@ -100,70 +100,28 @@ impl AspaDefinition {
         self.providers.iter().any(|p| p.provider() == self.customer)
     }
 
-    /// Returns true if the update would change this AspaDefinition
-    /// in any way. I.e. including a change in afiLimit for currently
-    /// configured ProviderAs entries.
-    pub fn needs_update(&self, update: &AspaProvidersUpdate) -> bool {
-        for removed in update.removed() {
-            match removed.afi_limit() {
-                None => {
-                    // if removed uses no limit, then we need an update if
-                    // we have any entry for the asn
-                    if self.providers.iter().any(|e| e.provider() == removed.provider()) {
-                        return true;
-                    }
-                }
-                Some(removed_limit) => {
-                    // removed wants a limit, so we only need to remove if
-                    // we currently have no limit, or the same limit.
-                    if let Some(existing) = self.providers.iter().find(|e| e.provider() == removed.provider()) {
-                        match existing.afi_limit() {
-                            None => {
-                                // we will need to shrink the afiLime
-                                return true;
-                            }
-                            Some(existing_limit) => {
-                                if removed_limit == existing_limit {
-                                    // there is a match to remove
-                                    return true;
-                                }
-                                // else -> there is no match to remove
-                            }
-                        }
-                    }
-                }
+    /// Returns true if this is a single AFI AspaDefinition. Technically,
```

Comment:
Your doc string and the method name contradict each other – the doc string suggests this returns true if there is only one AFI, the method name suggests it returns true if it has both AFIs.

Reply:
yeah, oops sorry.. refactored and then did not change this text

Reply:
fixed